plugins/generic/externalFeed/ExternalFeedBlockPlugin.inc.php

```diff
@@ -36,62 +39,67 @@ function getHideManagement() {
 	 * its category.
 	 * @return String name of plugin
 	 */
-	function getName() {
+	public function getName() {
 		return 'ExternalFeedBlockPlugin';
 	}
 
 	/**
 	 * Get the display name of this plugin.
 	 * @return String
 	 */
-	function getDisplayName() {
+	public function getDisplayName() {
 		return __('plugins.generic.externalFeed.block.displayName');
 	}
 
 	/**
 	 * Get a description of the plugin.
 	 */
-	function getDescription() {
+	public function getDescription() {
 		return __('plugins.generic.externalFeed.description');
 	}
 
 	/**
 	 * Get the external feed plugin
 	 * @return object
 	 */
-	function &getExternalFeedPlugin() {
-		$plugin =& PluginRegistry::getPlugin('generic', $this->parentPluginName);
-		return $plugin;
+	public function getExternalFeedPlugin() {
+		return $this->parentPlugin;
 	}
 
 	/**
 	 * Override the builtin to get the correct plugin path.
 	 * @return string
 	 */
-	function getPluginPath() {
-		$plugin =& $this->getExternalFeedPlugin();
+	public function getPluginPath() {
+		$plugin = $this->getExternalFeedPlugin();
 		return $plugin->getPluginPath();
 	}
 
+	/**
+	 * Override the builtin to get the correct template path.
+	 * @return string
+	 */
+	public function getTemplatePath($inCore = false) {
+		return $this->getExternalFeedPlugin()->getTemplatePath($inCore);
+	}
+
 	/**
 	 * Get the HTML contents for this block.
 	 * @param $templateMgr object
 	 * @param $request PKPRequest
 	 * @return $string
 	 */
-	function getContents(&$templateMgr, $request = null) {
-		$journal = $request->getJournal();
-		if (!$journal) return '';
+	public function getContents($templateMgr, $request = null) {
+		$context = $request->getContext();
+		if (!$context) return '';
 
-		$journalId = $journal->getId();
-		$plugin =& $this->getExternalFeedPlugin();
+		$plugin = $this->getExternalFeedPlugin();
 		if (!$plugin->getEnabled()) return '';
 
 		$requestedPage = $request->getRequestedPage();
 		$externalFeedDao = DAORegistry::getDAO('ExternalFeedDAO');
-		$plugin->import('simplepie.SimplePie');
```

Comment:
I don't think anything else in our apps uses SimplePie, does it? If not, then it's best to leave the SimplePie dependency inside the plugin, by adding a `composer.json` dependency in the plugin's installation directory.

How we fetch the dependencies depends on where the plugin lives. There are two ways this is currently done:
1. For plugins built into the main codebase, there are lines in `tools/buildpkg.sh` to run composer as part of the build process.
2. For plugins that live in their own repos, it's up to the plugin builder to fetch composer dependencies when building a release package to upload to github.

Currently this plugin lives in the `ojs` repo, but especially since it can be compatible with OMP with app-specific lingo removed, it would be better to move it to its own repo instead. I'd follow the second process.

Reply:
Meaning I add `composer.json` to the plugin folder and once the plugin is moved to its own repo I upload a release package (zip file) containing the vendor folder. right?

Reply:
Regarding `simplepie` library import. Okay to use `require_once(dirname(__FILE__) . '/vendor/autoload.php');` just like 
https://github.com/pkp/ojs/blob/master/plugins/paymethod/paypal/PaypalPaymentPlugin.inc.php#L17?

Reply:
Yes, I think that's the best way.